pkg/tempo/config.go

```diff
@@ -167,10 +199,65 @@ func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
 		retryConfig["max_elapsed_time"] = "60s"
 	}
 
-	otelMapStructure["exporters"] = map[string]interface{}{
-		"otlp": otlpExporter,
+	return otlpExporter, nil
+}
+
+// exporters builds one or multiple exporters from a remote_write block.
+// It also supports building an exporter from push_config.
+func (c *InstanceConfig) exporters() (map[string]interface{}, error) {
+	if len(c.RemoteWrite) == 0 {
+		otlpExporter, err := exporter(RemoteWriteConfig{
+			Endpoint:           c.PushConfig.Endpoint,
+			Compression:        c.PushConfig.Compression,
+			Insecure:           c.PushConfig.Insecure,
+			InsecureSkipVerify: c.PushConfig.InsecureSkipVerify,
+			BasicAuth:          c.PushConfig.BasicAuth,
+			SendingQueue:       c.PushConfig.SendingQueue,
+			RetryOnFailure:     c.PushConfig.RetryOnFailure,
+		})
+		return map[string]interface{}{
+			"otlp": otlpExporter,
+		}, err
 	}
 
+	exporters := map[string]interface{}{}
+	for i, remoteWriteConfig := range c.RemoteWrite {
+		exporter, err := exporter(remoteWriteConfig)
+		if err != nil {
+			return nil, err
+		}
+		exporterName := fmt.Sprintf("otlp/%d", i)
+		exporters[exporterName] = exporter
+	}
+	return exporters, nil
+}
+
+func (c *InstanceConfig) otelConfig() (*configmodels.Config, error) {
```

Comment:
Can we log a warning somewhere here if push_config is configured to note that it's deprecated in favor of remote_write?

Reply:
Added in 53d7f969f43d38d6a36a49da13a6604505da6bf9